From 3278b08ddfb3a1784acefe534396cd5b6147e673 Mon Sep 17 00:00:00 2001
From: Martin Abente Lahaye <tch@sugarlabs.org>
Date: Tue, 13 May 2014 10:40:35 -0400
Subject: [PATCH 1/2] Fix KeyValuesDialog cancel

KeyValuesDialog does not have a reponse object, as it is a derived
class of Gtk.Dialog, and its creation comes from a complete different
path comparing with other dialogs that has a response object.

Signed-off-by: Martin Abente Lahaye <tch@sugarlabs.org>
---
 src/jarabe/desktop/keydialog.py | 7 +------
 1 file changed, 1 insertion(+), 6 deletions(-)

diff --git a/src/jarabe/desktop/keydialog.py b/src/jarabe/desktop/keydialog.py
index 20cf846..f8ecf5a 100644
--- a/src/jarabe/desktop/keydialog.py
+++ b/src/jarabe/desktop/keydialog.py
@@ -277,14 +277,9 @@ class KeyValuesDialog(Gtk.Dialog):
                 key = child._get_key()
                 value = child._get_value()
                 self._add_key_value(key, value)
-
-            key_dialog.destroy()
             self._final_callback(self._settings,
                                  self._selected_auth_list)
-        else:
-            response = key_dialog.get_response_object()
-            response.set_error(CanceledKeyRequestError())
-            key_dialog.destroy()
+        key_dialog.destroy()
 
     def _add_key_value(self, key, value):
         for auth_param in self._selected_auth_list._params_list:
-- 
1.8.3.1

